### PR TITLE
#166404582 Users should be able to read and delete articles

### PR DESCRIPTION
**Pull request by Benkimeric**
#### What does this PR do?
- Implements user view all articles
- Implement user view one article
- Implement the ability to delete articles

#### Description of Task to be completed?
- Ensure an authenticated user should be able to delete an article if they own it
- Ensure unauthenticated users can only read articles, both one and list of all
- Write tests for all functionalities

#### How should this be manually tested?
- [x] **Clone the repository**

        $ git clone https://github.com/andela/ah-the-jedi-frontend.git

- [x] **Switch to the ah-the-jedi-frontend directory**

       $ cd ah-the-jedi-frontend/

- [x] **Switch to testing branch**

       $ git checkout ft-read-delete-articles-166404582

 
- [x] **Install the project dependencies**

      $  npm install

- [x] **Running the project**

       $ npm run start:dev

- [x] **Login a user**

       $ <your-localhost>/login

- [x] **View all articles**

       $ <your-localhost>/
    
- [x] **View one article --> Click an article read more.. text to open**

       $ <your-localhost>/articles/<article-slug>

#### What are the relevant pivotal tracker stories?
[ #166404582](https://www.pivotaltracker.com/story/show/166404582)
#### Any background context you want to add?
None
#### Screenshots
####View all and one article
successview

####Delete an article
delete

### CODE REVIEW

**Review by AlvinMugambi — APPROVED** [review]

**Review by AlvinMugambi — COMMENTED** [review]
Good work on implementing read article and very good designs

**Review by ElishaMisoi — APPROVED** [review]

**Review by AlvinMugambi — APPROVED** [review]

File: src/components/articles/Article.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <div className="flip-box">
+        <div className="flip-box-inner">
+          <div className="flip-box-front">
+            <Card>
+              <Card.Img variant="top" src={articleImage} className="homepage-card-image" />
+              <Card.Body>
+                <Card.Title>{title}</Card.Title>
+                <Card.Text>{description}</Card.Text>
+                <Card.Text>
+                  <span>By </span>
```
Comment (jkamz) [inline src/components/articles/Article.js:34]: Very nice implementation @Benkimeric . It is especially commendable that you were able to implement flipping cards. Also, very nice of you on implementing the feedback to include a space between 'By' and the user details.

File: src/helpers/index.js [hunk 2]
```diff
@@ -24,3 +26,14 @@ export const isLoggedIn = () => {
   }
   return false;
 };
+
```
Comment (Siffersari) [inline src/helpers/index.js:29]: Good job having this helper class to help the team validate the logged in user instances.

File: src/components/articles/Article.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              </Card.Body>
+            </Card>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+Article.propTypes = {
```
Comment (katherine95) [inline src/components/articles/Article.js:68]: Great job in ensuring props are validated.

**Review by katherine95 — APPROVED** [review]

**Review by jkamz — APPROVED** [review]

**Review by Siffersari — APPROVED** [review]
Good job implementing flipping card design. This is a nice touch.  Good integration with the Profile component by redirecting authors to their profiles when viewing the articles. 

articlesfetch

profilearticle

**Pull request merged**